lib/std/Thread/Condition.zig

```diff
@@ -152,9 +152,9 @@ pub const AtomicCondition = struct {
             @atomicStore(bool, &cond.pending, true, .SeqCst);
         }
 
-        mutex.unlock();
+        mutex.releaseDirect();
         waiter.data.wait();
-        mutex.lock();
+        _ = mutex.acquire();
```

Comment:
same argument as for line 155.

Reply:
My reasoning here was this: since we expect a locked mutex to be passed into the `wait` method of the condition variable, we already have an instance of `Held` for that mutex in the calling scope. I thought to change the interface for `wait` to return the `Held` instance, but that felt confusing to me, because we already have an instance in the calling scope associated with the mutex. Very likely we have code like this:
```
// the mutex exists somewhere, maybe outside this scope
const held = mutex.acquire();
defer held.release();
while(! some_condition) {
  condvar.wait(&mutex);
}
```
Since the mutex is guaranteed to be locked when `wait` returns, it is semantically correct to release the mutex using the `held` instance. At least I think so.

Now if I make `wait` return the `Held` instance, I would have two instances pointing to the same mutex, which is confusing at best. Let me know if that makes sense.

Reply:
> Let me know if that makes sense.

yes. looks good. The overall code in `Thread` looks good, but there is no nice overview how stuff works for good code navigation. So it took me a while to figure things out.